tools/buildmgr/cbuildgen/config/GCC.11.2.1.cmake

```diff
@@ -15,6 +15,77 @@ set(CC ${TOOLCHAIN_ROOT}/${PREFIX}gcc${EXT})
 set(CXX ${TOOLCHAIN_ROOT}/${PREFIX}g++${EXT})
 set(OC ${TOOLCHAIN_ROOT}/${PREFIX}objcopy${EXT})
 
+# Helpers
+
+function(cbuild_set_defines lang defines)
+  set(TMP_DEFINES)
+  foreach(DEFINE ${${defines}})
+    string(REPLACE "\"" "\\\"" ENTRY ${DEFINE})
+    string(REGEX REPLACE "=.*" "" KEY ${ENTRY})
+    if (KEY STREQUAL ENTRY)
+      set(VALUE "1")
+    else()
+      string(REGEX REPLACE ".*=" "" VALUE ${ENTRY})
+    endif()
+    if(${lang} STREQUAL "AS_LEG")
+      string(APPEND TMP_DEFINES "--defsym ${KEY}=${VALUE} ")
+    elseif(${lang} STREQUAL "AS_GNU")
+      string(APPEND TMP_DEFINES "-Wa,-defsym,\"${KEY}=${VALUE}\" ")
+    else()
+      string(APPEND TMP_DEFINES "-D${ENTRY} ")
+    endif()
+  endforeach()
+  set(${defines} ${TMP_DEFINES} PARENT_SCOPE)
+endfunction()
+
+set(OPTIMIZE_VALUES       "debug" "none" "balanced" "size" "speed")
+set(OPTIMIZE_CC_FLAGS     "-Og"   "-O0"  "-O2"      "-Oz" "-O3")
```

Comment:
What is the rational for using `-Oz` for `size`? Maybe it would be safer to use `-Os` which is less aggressive, similarly about using `O3` and not `Ofast` for `speed`?

Reply:
I have thought I can use the best size optimization possible.
But let's shoot for less-aggressive size optimization.